security/WFLY-14922_Remove_PicketLink_Subsystem.adoc

```diff
@@ -0,0 +1,125 @@
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1750[EAP7-1750 - Remove PicketLink Subsystem]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [x] QE
+
+As a removal RFE, this does not require new feature test development. However both teams will need
+to review existing testsuites for tests dependent on the PicketLink subsystem. Identified tests will
+either need to be deleted or they will need to be converted to use the new approach.
+
+=== Affected Projects or Components
+
+* WildFly
+* Keycloak
+
+=== Other Interested Projects
+
+* [x] External Testsuites
+
+* [x] Quickstarts
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+Since this is a removal task, this section will describe the steps that will be needed as opposed to the
+set of requirements.
+
+=== Hard Requirements
+
+* The PicketLink subsystem needs to be converted to a skeleton subsystem. (It needs to be a skeleton subsystem
+instead of being removed completely so that older domains can still be managed.)
+
+* Any affected tests will need to be removed from WildFly.
+
+* Any documentation that references PicketLink will need to be removed. The main section in the WildFly documentation
+that is affected is the "WS-Trust and STS section".
+
+* The `org.keycloak.keycloak-saml-wildfly-elytron-adapter` module that's installed by the Keycloak SAML Elytron
+adapter needs to be updated to remove the `org.picketbox` module from its dependencies (the `org.picketbox` dependency
+does not appear to be used, it just needs to be removed).
+
+=== Nice-to-Have Requirements
+
+N/A
+
+=== Non-Requirements
+
+* The Keycloak SAML Elytron adapter does not ship with WildFly. It needs to be installed separately by the
+user. Thus, migrating from SSO with SAML using PicketLink to SSO with SAML via the Keycloak adapter won't be
+something that's handled by the server migration tool. The user will need to install the Keycloak adapter on
+their own.
+
+== Implementation Plan
+
+N/A
+
+== Test Plan
+
+To be discussed.
+
+The Keycloak SAML Elytron adapter already exists and is already used and tested today. Testing will
+be needed to make sure no issues arise after removing the `org.picketbox` module dependency
+from the `org.keycloak.keycloak-saml-wildfly-elytron-adapter` module.
+
+== Community Documentation
+
+In addition to removing any references to PicketLink from the community documentation,
+a "migration" article will be added to indicate that the Keycloak SAML Elytron adapter
+should be used to configure SSO with SAML for WildFly. Users should refer to the Keycloak
```

Comment:
Is the Keycloak SAML Elytron adapter usable with other SAML identity providers than Keycloak? It is possible to use PicketLink SP with non-PicketLink IDPs (and users do so). In case this is not possible with EAP Keycloak SP/adapter, such fact should be clearly visible.

Reply:
@OndrejKotek I'll double check but it does seem to be possible to use the Keycloak SAML Elytron adapter with other SAML identity providers (e.g., see https://developers.redhat.com/blog/2019/08/27/using-keycloak-instead-of-picketlink-for-saml-based-authentication).

Reply:
@OndrejKotek I looked into this and confirmed that this is possible.